api/v1alpha1/sandbox_types.go

```diff
@@ -58,6 +83,16 @@ type PodTemplate struct {
 	ObjectMeta PodMetadata `json:"metadata" protobuf:"bytes,3,opt,name=metadata"`
 }
 
+type PersistentVolumeClaimTemplate struct {
+	// Metadata is the Pod's metadata. Only labels and annotations are used.
+	// +kubebuilder:validation:Optional
+	ObjectMeta PVCMetadata `json:"metadata" protobuf:"bytes,3,opt,name=metadata"`
```

Comment:
Why not just use `metav1.ObjectMeta`, instead of defining our own over and over? We can use validation to prevent users from setting fields that we don't want them to set. Similarly, StatefulSet spec embeds `PodTemplateSpec` and `PersistentVolumeClaim`, both contain `metav1.ObjectMeta`, instead of trying to define a custom type of object meta.

Reply:
Original reason for this pattern for podTemplate was https://github.com/kubernetes-sigs/agent-sandbox/issues/38 

I will revert to using PersistentVolumeClaim as is and if we run into issues we can explore further.